apps/managedidentity/managedidentity.go

```diff
@@ -173,17 +280,104 @@ func createIMDSAuthRequest(ctx context.Context, id ID, resource string, claims s
 	return req, nil
 }
 
-func (client Client) getTokenForRequest(req *http.Request) (accesstokens.TokenResponse, error) {
+func createAzureArcAuthRequest(ctx context.Context, id ID, resource string, claims string) (*http.Request, error) {
+	identityEndpoint, _ := getAzureArcEnvironmentVariables()
+	var msiEndpoint *url.URL
+
+	if identityEndpoint == "" {
+		identityEndpoint = azureArcEndpoint
+	}
+
+	if _, ok := id.(systemAssignedValue); !ok {
+		return nil, errors.New("Azure Arc doesn't support specifying a user-assigned managed identity at runtime")
+	}
+
+	identityEndpoint = strings.Replace(identityEndpoint, "localhost", "127.0.0.1", -1)
```

Comment:
Why do you need to replace this string?

Reply:
I saw MSAL JS was doing this so I added it here: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/f0c5f0250741cfd673538ce51c19a345fdf9be75/lib/msal-node/src/client/ManagedIdentitySources/AzureArc.ts#L192

Reply:
Hmm, if you get the endpoint from the env variable I think you should not alter that

Reply:
Ok sounds good will update :)